lib/widgets/modular_widgets/wallet_connect_widgets/wallet_connect_pairing_list_card.dart

```diff
@@ -0,0 +1,153 @@
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CardScaffold(
+      title: _kWidgetTitle,
+      description: _kWidgetDescription,
+      childBuilder: () => _getCardBody(),
+      onRefreshPressed: () => _pairingsBloc.refreshResults(),
+    );
+  }
+
+  Widget _getCardBody() {
+    return _buildPairingsTable();
+  }
+
+  Widget _buildPairingsTable() {
+    return InfiniteScrollTable<PairingInfo>(
+      disposeBloc: false,
+      bloc: _pairingsBloc,
+      headerColumns: const [
+        InfiniteScrollTableHeaderColumn(
+          columnName: 'Name',
+        ),
+        InfiniteScrollTableHeaderColumn(
+          columnName: 'URL',
+          flex: 2,
+        ),
+        InfiniteScrollTableHeaderColumn(
+          columnName: 'Topic',
+        ),
+        InfiniteScrollTableHeaderColumn(
+          columnName: 'Expiration',
+        ),
+        InfiniteScrollTableHeaderColumn(
+          columnName: 'Active',
+        ),
+        InfiniteScrollTableHeaderColumn(
+          columnName: '',
+        ),
+      ],
+      generateRowCells: (pairingInfo, bool isSelected) {
+        return [
+          InfiniteScrollTableCell.withText(
+            context,
+            pairingInfo.peerMetadata?.name ?? 'Empty',
+          ),
+          InfiniteScrollTableCell(
+            _buildTableUrlWidget(pairingInfo),
+            flex: 2,
+          ),
+          isSelected
+              ? InfiniteScrollTableCell.withMarquee(
+                  pairingInfo.topic,
+                )
+              : InfiniteScrollTableCell.withText(
+                  context,
+                  pairingInfo.topic.short,
+                ),
+          InfiniteScrollTableCell.withText(
+            context,
+            _formatExpiryDateTime(pairingInfo.expiry).toString(),
+          ),
+          InfiniteScrollTableCell.withText(
+            context,
+            pairingInfo.active ? 'Yes' : 'No',
+          ),
+          InfiniteScrollTableCell(
+            _buildDeactivatePairingIcon(pairingInfo),
+          ),
+        ];
+      },
+    );
+  }
+
+  ClearIcon _buildDeactivatePairingIcon(PairingInfo pairingInfo) {
+    return ClearIcon(
+      onPressed: () => _onDeactivatePairingIconPressed(pairingInfo),
+      context: context,
+    );
+  }
+
+  Row _buildTableUrlWidget(PairingInfo pairingInfo) {
+    return Row(
+      children: [
+        Text(pairingInfo.peerMetadata?.url ?? 'Empty'),
+        Visibility(
+          visible: pairingInfo.peerMetadata?.url != null,
+          child: LinkIcon(
+            url: pairingInfo.peerMetadata?.url ?? 'Empty',
+          ),
+        ),
+      ],
+    );
+  }
+
+  String _formatExpiryDateTime(int expirySeconds) {
+    final expiryDateTime =
+        DateTime.fromMillisecondsSinceEpoch(expirySeconds * 1000);
+
+    return DateFormat('MMM dd, y HH:mm:ss').format(expiryDateTime);
+  }
+
+  Future<void> _onDeactivatePairingIconPressed(PairingInfo pairingInfo) async {
```

Comment:
Should a pairing's sessions also be disconnected when a pairing is deactivated? I don't think WalletConnect does that automatically but it could make it easier for dapps to manage sessions, so that they don't need to necessarily use the Pairing API to listen to the `pairing_delete` event to manage their sessions.